golang/lib/services/shared_directory.go

```diff
@@ -1,21 +1,50 @@
 package services
 
+import (
+	"github.com/kurtosis-tech/kurtosis-client/golang/lib/services/shared_file_object"
+	"github.com/palantir/stacktrace"
+	"os"
+	"path/filepath"
+)
+
+//TODO add documentation
 type SharedDirectory struct {
 	//The service directory absolute path in the container where this code is running
-	absFilepathOnThisContainer string
+	sharedDirectoryMountpathOnThisContainer string
 	//The service directory absolute path in the service container
-	absFilepathOnServiceContainer string
+	sharedDirectoryMountpathOnServiceContainer string
 }
 
 func NewSharedDirectory(absFilepathOnThisContainer string, absFilepathOnServiceContainer string) *SharedDirectory {
-	return &SharedDirectory{absFilepathOnThisContainer: absFilepathOnThisContainer, absFilepathOnServiceContainer: absFilepathOnServiceContainer}
+	return &SharedDirectory{sharedDirectoryMountpathOnThisContainer: absFilepathOnThisContainer, sharedDirectoryMountpathOnServiceContainer: absFilepathOnServiceContainer}
 }
 
-func (s SharedDirectory) GetAbsFilepathOnThisContainer() string {
-	return s.absFilepathOnThisContainer
+//TODO add documentation
+func (s SharedDirectory) GetSharedDirectoryMountpathOnThisContainer() string {
+	return s.sharedDirectoryMountpathOnThisContainer
 }
 
-func (s SharedDirectory) GetAbsFilepathOnServiceContainer() string {
-	return s.absFilepathOnServiceContainer
+//TODO add documentation
+func (s SharedDirectory) GetSharedDirectoryMountpathOnServiceContainer() string {
+	return s.sharedDirectoryMountpathOnServiceContainer
 }
 
+//TODO add documentation
+func (s SharedDirectory) GetSharedFileObject(fileName string) (*shared_file_object.SharedFileObject, error) {
+
+	absFilepathOnThisContainer := filepath.Join(s.sharedDirectoryMountpathOnThisContainer, fileName)
+
+	if _, err := os.Stat(absFilepathOnThisContainer); os.IsNotExist(err) {
```

Comment:
Per Slack, I'd strongly suggest making this method "dumber" - i.e. maybe just calling the returned object `SharedDirectoryPath`, that just has two objects